crates/autopilot/src/solvable_orders.rs

```diff
@@ -210,24 +219,34 @@ impl SolvableOrdersCache {
         }
 
         let orders = orders_with_balance(orders, &new_balances, self.ethflow_contract_address);
-        counter.checkpoint("insufficient_balance", &orders);
+        let removed = counter.checkpoint("insufficient_balance", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Invalid);
 
         let orders = filter_dust_orders(orders, &new_balances, self.ethflow_contract_address);
-        counter.checkpoint("dust_order", &orders);
+        let removed = counter.checkpoint("dust_order", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Filtered);
 
         // create auction
         let (orders, prices) = get_orders_with_native_prices(
             orders.clone(),
             &self.native_price_estimator,
             self.metrics,
         );
-        counter.checkpoint("missing_price", &orders);
+        let removed = counter.checkpoint("missing_price", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Filtered);
 
         let orders = filter_mispriced_limit_orders(orders, &prices, &self.limit_order_price_factor);
-        counter.checkpoint("out_of_market", &orders);
+        let removed = counter.checkpoint("out_of_market", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Filtered);
 
         let orders = apply_fee_objective_scaling_factor(orders, &self.fee_objective_scaling_factor);
-        counter.checkpoint("fee_scaling_overflow", &orders);
+        let removed = counter.checkpoint("fee_scaling_overflow", &orders);
+        self.database
+            .store_order_events(removed, OrderEventLabel::Filtered);
 
         let auction = Auction {
```

Comment:
`counter.record()` logs warnings for "uncheckpointed" orders. Should we also have something catch-all like:

```rust
        let removed = counter.record(&auction.orders);
        self.database
            .store_order_events(removed, OrderEventLabel::Filtered);
```

Reply:
[d0d0770](https://github.com/cowprotocol/services/pull/1641/commits/d0d0770fc23ad39fcb6ff2c25eaa2f242aa1adf9)